### PR TITLE
chore: introduce wrapped JSON schema identifier (1.5.x backport)

### LINKED ISSUE #3914: Add $id property to the JSON schema URI in the DataShape
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[*] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
We need to follow this up with a change that includes the syndesis JSON schema URI in the data shape so that `ResponseCustomizer` can detect the need to unwrap the response here:

https://github.com/syndesisio/syndesis/blob/9e99679cd43adf3cfe171d4218ec4f31b2ea83a0/app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java#L64

That means we need to add `$id`[1] property to the unified JSON schemas. Similar to how the WebHook connector does it:

https://github.com/syndesisio/syndesis/blob/9e99679cd43adf3cfe171d4218ec4f31b2ea83a0/app/connector/webhook/src/main/java/io/syndesis/connector/webhook/WebhookConnectorCustomizer.java#L39

https://github.com/syndesisio/syndesis/blob/9e99679cd43adf3cfe171d4218ec4f31b2ea83a0/app/connector/webhook/src/main/java/io/syndesis/connector/webhook/WebhookConnectorCustomizer.java#L82

[1] https://json-schema.org/understanding-json-schema/structuring.html#the-id-property

### PR DESCRIPTION
**Pull request by zregvart**
This introduces the wrapped JSON schema identifier of `io:syndesis:wrapped`, we use it to detect if the response needs to be _wrapped_ i.e. convert `{"hello": "world"}` to `{"body": {"hello: "world"}}`.

Also tries to stay backward compatible. Because of this there is a possibility to detect some payloads as wrapped, and that's why it's better to rely on the schema identifier.

Fixes #3914

(cherry picked from commit 2bc2488680b716bce80068154979308fb08d1d98)

### CODE REVIEW

File: app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java [hunk 1]
```diff
@@ -57,6 +57,37 @@ public void setOutputDataShape(final DataShape dataShape) {
         isUnifiedDataShape = isUnifiedDataShape(dataShape);
     }
```
Comment (KurtStam) [inline app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java]: At what point can we remove checking the legacy way? Should we add a deprecate warning?

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java [hunk 2]
```diff
@@ -288,6 +288,8 @@ private static DataShape unifiedJsonSchema(final String name, final String descr
         }
 
         final ObjectNode unifiedSchema = JsonSchemaHelper.newJsonObjectSchema();
+        // we're using Draft 4, in Draft 6+ this should be `$id`
```
Comment (KurtStam) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: Can we move to Draft 6+ now?

Comment (zregvart) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: up to Atlasmap I think, and for another PR

File: app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java (hunk 1 above)
Comment (zregvart) [reply app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java]: The data shapes could be baked into Integration (like the model), I'd prefer to drop it, but I'm unsure if the integrations stop working at that point. It depends on how many integrations are living with the data shapes in JSON schema without the `id`.

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java (hunk 2 above)
Comment (KurtStam) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: How about opening and issue in AltasMap for it and reference that here?

File: app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java (hunk 1 above)
Comment (KurtStam) [reply app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java]: You're saying to support legacy forever?

Comment (zregvart) [reply app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java]: @KurtStam what's your suggestion?

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java (hunk 2 above)
Comment (zregvart) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: Actually it us:

https://github.com/atlasmap/atlasmap/blob/70020a7f78feec388efb289352dc0593ef4eeee4/runtime/modules/json/core/src/main/java/io/atlasmap/json/inspect/SchemaInspector.java#L90

https://github.com/atlasmap/atlasmap/blob/70020a7f78feec388efb289352dc0593ef4eeee4/runtime/modules/json/core/src/test/resources/inspect/schema/flatprimitive-base-unrooted.json#L2

Comment (zregvart) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: Right, Mr. Smartypants

File: app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java (hunk 1 above)
Comment (zregvart) [reply app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java]: Actually, I just realized we added this in 1.5, so removing it won't impact anyone really. I'm going to drop it.

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java (hunk 2 above)
Comment (KurtStam) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/UnifiedJsonDataShapeGenerator.java]: Capt Obvious to you

**Pull request merged**